ts-client/src/dlmm/index.ts

```diff
@@ -4851,6 +4818,38 @@ export class DLMM {
     return totalWithdrawAmount;
   }
 
+  /**
+   *
+   * @param swapInitiator Address of the swap initiator
+   * @returns
+   */
+  public isSwapDisabled(swapInitiator: PublicKey) {
+    if (this.lbPair.status == PairStatus.Disabled) {
+      return true;
+    }
+
+    if (this.lbPair.pairType == PairType.Permissioned) {
+      const currentPoint =
+        this.lbPair.activationType == ActivationType.Slot
+          ? this.clock.slot
+          : this.clock.unixTimestamp;
+
+      const preActivationSwapPoint = this.lbPair.activationPoint.sub(
+        this.lbPair.preActivationDuration
+      );
+
+      const activationPoint =
+        !this.lbPair.whitelistedWallet.equals(PublicKey.default) &&
+        this.lbPair.whitelistedWallet.equals(swapInitiator)
+          ? preActivationSwapPoint
+          : this.lbPair.activationPoint;
+
+      if (currentPoint < activationPoint) {
+        return true;
+      }
+    }
```

Comment:
why not return false at the end of function

Reply:
Fixed